rct229/ruletest_engine/ruletest_engine.py

```diff
@@ -337,6 +343,208 @@ def run_section_tests(test_json_name: str, ruleset_doc: str):
     return all_tests_pass
 
 
+def generate_software_test_report(ruleset, section_list, output_json_path):
+    """Runs list of rule test JSONs and aggregates them into a ashrae901_2019_detail_report
+
+    Parameters
+    ----------
+    ruleset: string
+
+        Name of the ruleset (e.g., 'ashrae9012019')
+
+    section_list : list
+
+        List of test JSON directorys in 'test_jsons/[MY_STANDARD]' directory. (e.g., ['section5', 'section6'])
+
+    output_json_path: str
+
+        output_dir: str - directory in which you want the ashrae901_2019_software_testing_report.json to appear
+
+
+    """
+
+    # Initialize report dictionary in which to
+    if ruleset == "ashrae9012019":
+        report_dict = ASHRAE9012019SoftwareTestReport()
+        report_dict.initialize_ruleset_report()
+    else:
+        raise Exception(f"Ruleset '{ruleset}' has no default list of section tests.")
+
+    if section_list is None:
+
+        if ruleset == "ashrae902019":
+            section_list = RuleSetTest.ASHRAE9012019_TEST_LIST
+        else:
+            raise Exception(
+                f"Ruleset '{ruleset}' has no default list of section tests."
+            )
+
+    # Master list of RCT engine outcomes, used to populate report.
+    rct_outcomes = generate_rct_outcomes_list_from_section_list(section_list, ruleset)
+
+    # Generate
+    report_dict.generate(rct_outcomes, output_json_path)
+
+
+def generate_rct_outcomes_list_from_section_list(section_list, ruleset):
+    """Runs all the ruletest JSONs for every section in section_list for a given ruleset. Returns the aggregated
+    results as a dictionary that can be used in the generate function for ashrae901_2019_software_test_report
+
+    Parameters
+    ----------
+    section_list : list
+
+        List of test JSON directorys in 'test_jsons/[MY_STANDARD]' directory. (e.g., ['section5', 'section6'])
+
+    ruleset: string
+
+        Name of the ruleset (e.g., 'ashrae9012019')
+
+    Returns
+    ----------
+    rct_outcomes_dict: dict
+
+        The dictionary with aggregated results to used in the generate function for the
+        ashrae901_2019_software_test_report
+
+    """
+
+    # Master list of RCT engine outcomes and invalid RMR messages used to populate starting point for an RCTReport.
+    # Initialize them here
+    rct_outcomes_list = []
+    invalid_rmr_messages = []
+
+    # Maps section lists to their titles TODO: resolve 19 vs 23
```

Comment:
This is something I could use help with.

Reply:
Rough categories for each section per Weili on call:

18 - HVAC-BaselineHVAC
19 - HVAC-Airside
23 - HVAC-SystemSpecificRequirements